test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -0,0 +1,316 @@
+	writeHistories map[bool][]map[string][]time.Time
+}
+
+func newFightTest(t *testing.T, loopbackConfig *rest.Config, size int) *fightTest {
+	now := time.Now()
+	ft := &fightTest{
+		t:              t,
+		ctx:            context.Background(),
+		loopbackConfig: loopbackConfig,
+		size:           size,
+		stopCh:         make(chan struct{}),
+		now:            now,
+		clk:            clock.NewFakeClock(now),
+		ctlrs: map[bool][]utilfc.TestableInterface{
+			false: make([]utilfc.TestableInterface, size),
+			true:  make([]utilfc.TestableInterface, size)},
+		notifiedRVs: map[bool][]map[string]string{
+			false: make([]map[string]string, size),
+			true:  make([]map[string]string, size)},
+		writeHistories: map[bool][]map[string][]time.Time{
+			false: make([]map[string][]time.Time, size),
+			true:  make([]map[string][]time.Time, size)},
+	}
+	ft.foreach(func(invert bool, i int) {
+		ft.rvMutex.Lock()
+		defer ft.rvMutex.Unlock()
+		ft.notifiedRVs[invert][i] = map[string]string{}
+		ft.writeHistories[invert][i] = map[string][]time.Time{}
+	})
+	return ft
+}
+
+func (ft *fightTest) createController(invert bool, i int) {
+	myConfig := rest.CopyConfig(ft.loopbackConfig)
+	myConfig = rest.AddUserAgent(myConfig, fmt.Sprintf("invert=%v, i=%d", invert, i))
+	myClientset := clientset.NewForConfigOrDie(myConfig)
+	fcIfc := myClientset.FlowcontrolV1beta1()
+	if ft.lastRVs == nil {
+		var err error
+		ft.lastRVs, err = getResourceVersionOfEachFlowSchema(ft.t, fcIfc.FlowSchemas())
+		if err != nil {
+			ft.t.Fatal(err)
+		}
+	}
+	informerFactory := informers.NewSharedInformerFactory(myClientset, 0)
+	fieldMgr := utilfc.ConfigConsumerAsFieldManager
+	foundToDangling := func(found bool) bool { return !found }
+	if invert {
+		fieldMgr = fieldMgr + "x"
+		foundToDangling = func(found bool) bool { return found }
+	}
+	ctlr := utilfc.NewTestable(utilfc.TestableConfig{
+		Name:  fmt.Sprintf("Controller%d[invert=%v]", i, invert),
+		Clock: ft.clk,
+		FinishHandlingNotification: func(wq workqueue.RateLimitingInterface, obj interface{}) {
+			ft.t.Logf("For invert=%v, i=%v, notified of %#+v", invert, i, obj)
+			obj = peel(obj)
+			switch typed := obj.(type) {
+			case *flowcontrol.FlowSchema:
+				key, _ := cache.MetaNamespaceKeyFunc(obj)
+				ft.rvMutex.Lock()
+				defer ft.rvMutex.Unlock()
+				ft.notifiedRVs[invert][i][key] = typed.ResourceVersion
+			}
+		},
+		AsFieldManager:         fieldMgr,
+		FoundToDangling:        foundToDangling,
+		InformerFactory:        informerFactory,
+		FlowcontrolClient:      fcIfc,
+		ServerConcurrencyLimit: 200,         // server concurrency limit
+		RequestWaitLimit:       time.Minute, // request wait limit
+		ObsPairGenerator:       metrics.PriorityLevelConcurrencyObserverPairGenerator,
+		QueueSetFactory:        fqtesting.NewNoRestraintFactory(),
+	})
+	ft.ctlrs[invert][i] = ctlr
+	informerFactory.Start(ft.stopCh)
+	if ctlr.WaitForCacheSync(ft.stopCh) {
+		ft.t.Logf("Achieved initial sync for invert=%v, i=%v", invert, i)
```

Comment:
I don't think this log is particularly useful - you Fatalf in the else branch anyway...

let's remove it.